internal/config/config.go

```diff
@@ -64,7 +64,7 @@ func readFrom(fs afero.Fs, path string) (c config.Config, err error) {
 
 	// kubebuilder v1 omitted version, so default to v1
 	if c.Version == "" {
-		c.Version = config.Version1
+		return config.Config{}, fmt.Errorf("project version key `version` is empty or does not exist in %s", path)
```

Comment:
IMO we need a PR to ADD the migration guide v2 to v3
and another one to do these changes.